// react-resplit/dist/resplit.es.js
import et, { forwardRef as ge, Children as Ee, isValidElement as xe, createElement as Fe, cloneElement as tt, Fragment as jt, createContext as rt, useContext as nt, useEffect as Ne, useLayoutEffect as $t, useId as At, useRef as we, useState as kt, useCallback as j, useMemo as Ze } from "react";
var Ie = { exports: {} }, he = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Xe;
function Dt() {
  if (Xe)
    return he;
  Xe = 1;
  var a = et, i = Symbol.for("react.element"), u = Symbol.for("react.fragment"), d = Object.prototype.hasOwnProperty, R = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, S = { key: !0, ref: !0, __self: !0, __source: !0 };
  function x(T, b, _) {
    var y, c = {}, I = null, g = null;
    _ !== void 0 && (I = "" + _), b.key !== void 0 && (I = "" + b.key), b.ref !== void 0 && (g = b.ref);
    for (y in b)
      d.call(b, y) && !S.hasOwnProperty(y) && (c[y] = b[y]);
    if (T && T.defaultProps)
      for (y in b = T.defaultProps, b)
        c[y] === void 0 && (c[y] = b[y]);
    return { $$typeof: i, type: T, key: I, ref: g, props: c, _owner: R.current };
  }
  return he.Fragment = u, he.jsx = x, he.jsxs = x, he;
}
var ye = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Qe;
function Ft() {
  return Qe || (Qe = 1, process.env.NODE_ENV !== "production" && function() {
    var a = et, i = Symbol.for("react.element"), u = Symbol.for("react.portal"), d = Symbol.for("react.fragment"), R = Symbol.for("react.strict_mode"), S = Symbol.for("react.profiler"), x = Symbol.for("react.provider"), T = Symbol.for("react.context"), b = Symbol.for("react.forward_ref"), _ = Symbol.for("react.suspense"), y = Symbol.for("react.suspense_list"), c = Symbol.for("react.memo"), I = Symbol.for("react.lazy"), g = Symbol.for("react.offscreen"), U = Symbol.iterator, G = "@@iterator";
    function N(e) {
      if (e === null || typeof e != "object")
        return null;
      var t = U && e[U] || e[G];
      return typeof t == "function" ? t : null;
    }
    var k = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function P(e) {
      {
        for (var t = arguments.length, r = new Array(t > 1 ? t - 1 : 0), l = 1; l < t; l++)
          r[l - 1] = arguments[l];
        ne("error", e, r);
      }
    }
    function ne(e, t, r) {
      {
        var l = k.ReactDebugCurrentFrame, h = l.getStackAddendum();
        h !== "" && (t += "%s", r = r.concat([h]));
        var E = r.map(function(p) {
          return String(p);
        });
        E.unshift("Warning: " + t), Function.prototype.apply.call(console[e], console, E);
      }
    }
    var X = !1, oe = !1, ce = !1, W = !1, ae = !1, ie;
    ie = Symbol.for("react.module.reference");
    function Re(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === d || e === S || ae || e === R || e === _ || e === y || W || e === g || X || oe || ce || typeof e == "object" && e !== null && (e.$$typeof === I || e.$$typeof === c || e.$$typeof === x || e.$$typeof === T || e.$$typeof === b || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ie || e.getModuleId !== void 0));
    }
    function _e(e, t, r) {
      var l = e.displayName;
      if (l)
        return l;
      var h = t.displayName || t.name || "";
      return h !== "" ? r + "(" + h + ")" : r;
    }
    function fe(e) {
      return e.displayName || "Context";
    }
    function L(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && P("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case d:
          return "Fragment";
        case u:
          return "Portal";
        case S:
          return "Profiler";
        case R:
          return "StrictMode";
        case _:
          return "Suspense";
        case y:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case T:
            var t = e;
            return fe(t) + ".Consumer";
          case x:
            var r = e;
            return fe(r._context) + ".Provider";
          case b:
            return _e(e, e.render, "ForwardRef");
          case c:
            var l = e.displayName || null;
            return l !== null ? l : L(e.type) || "Memo";
          case I: {
            var h = e, E = h._payload, p = h._init;
            try {
              return L(p(E));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var B = Object.assign, Q = 0, Se, Pe, o, n, s, v, m;
    function D() {
    }
    D.__reactDisabledLog = !0;
    function O() {
      {
        if (Q === 0) {
          Se = console.log, Pe = console.info, o = console.warn, n = console.error, s = console.group, v = console.groupCollapsed, m = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: D,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        Q++;
      }
    }
    function M() {
      {
        if (Q--, Q === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: B({}, e, {
              value: Se
            }),
            info: B({}, e, {
              value: Pe
            }),
            warn: B({}, e, {
              value: o
            }),
            error: B({}, e, {
              value: n
            }),
            group: B({}, e, {
              value: s
            }),
            groupCollapsed: B({}, e, {
              value: v
            }),
            groupEnd: B({}, e, {
              value: m
            })
          });
        }
        Q < 0 && P("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var A = k.ReactCurrentDispatcher, Y;
    function $(e, t, r) {
      {
        if (Y === void 0)
          try {
            throw Error();
          } catch (h) {
            var l = h.stack.trim().match(/\n( *(at )?)/);
            Y = l && l[1] || "";
          }
        return `
` + Y + e;
      }
    }
    var K = !1, ee;
    {
      var de = typeof WeakMap == "function" ? WeakMap : Map;
      ee = new de();
    }
    function H(e, t) {
      if (!e || K)
        return "";
      {
        var r = ee.get(e);
        if (r !== void 0)
          return r;
      }
      var l;
      K = !0;
      var h = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var E;
      E = A.current, A.current = null, O();
      try {
        if (t) {
          var p = function() {
            throw Error();
          };
          if (Object.defineProperty(p.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(p, []);
            } catch (J) {
              l = J;
            }
            Reflect.construct(e, [], p);
          } else {
            try {
              p.call();
            } catch (J) {
              l = J;
            }
            e.call(p.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (J) {
            l = J;
          }
          e();
        }
      } catch (J) {
        if (J && l && typeof J.stack == "string") {
          for (var f = J.stack.split(`
`), F = l.stack.split(`
`), C = f.length - 1, w = F.length - 1; C >= 1 && w >= 0 && f[C] !== F[w]; )
            w--;
          for (; C >= 1 && w >= 0; C--, w--)
            if (f[C] !== F[w]) {
              if (C !== 1 || w !== 1)
                do
                  if (C--, w--, w < 0 || f[C] !== F[w]) {
                    var V = `
` + f[C].replace(" at new ", " at ");
                    return e.displayName && V.includes("<anonymous>") && (V = V.replace("<anonymous>", e.displayName)), typeof e == "function" && ee.set(e, V), V;
                  }
                while (C >= 1 && w >= 0);
              break;
            }
        }
      } finally {
        K = !1, A.current = E, M(), Error.prepareStackTrace = h;
      }
      var ue = e ? e.displayName || e.name : "", Ke = ue ? $(ue) : "";
      return typeof e == "function" && ee.set(e, Ke), Ke;
    }
    function z(e, t, r) {
      return H(e, !1);
    }
    function se(e) {
      var t = e.prototype;
      return !!(t && t.isReactComponent);
    }
    function te(e, t, r) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return H(e, se(e));
      if (typeof e == "string")
        return $(e);
      switch (e) {
        case _:
          return $("Suspense");
        case y:
          return $("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case b:
            return z(e.render);
          case c:
            return te(e.type, t, r);
          case I: {
            var l = e, h = l._payload, E = l._init;
            try {
              return te(E(h), t, r);
            } catch {
            }
          }
        }
      return "";
    }
    var Z = Object.prototype.hasOwnProperty, pe = {}, ve = k.ReactDebugCurrentFrame;
    function q(e) {
      if (e) {
        var t = e._owner, r = te(e.type, e._source, t ? t.type : null);
        ve.setExtraStackFrame(r);
      } else
        ve.setExtraStackFrame(null);
    }
    function ct(e, t, r, l, h) {
      {
        var E = Function.call.bind(Z);
        for (var p in e)
          if (E(e, p)) {
            var f = void 0;
            try {
              if (typeof e[p] != "function") {
                var F = Error((l || "React class") + ": " + r + " type `" + p + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[p] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw F.name = "Invariant Violation", F;
              }
              f = e[p](t, p, l, r, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (C) {
              f = C;
            }
            f && !(f instanceof Error) && (q(h), P("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", l || "React class", r, p, typeof f), q(null)), f instanceof Error && !(f.message in pe) && (pe[f.message] = !0, q(h), P("Failed %s type: %s", r, f.message), q(null));
          }
      }
    }
    var ft = Array.isArray;
    function je(e) {
      return ft(e);
    }
    function dt(e) {
      {
        var t = typeof Symbol == "function" && Symbol.toStringTag, r = t && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return r;
      }
    }
    function pt(e) {
      try {
        return Me(e), !1;
      } catch {
        return !0;
      }
    }
    function Me(e) {
      return "" + e;
    }
    function Ye(e) {
      if (pt(e))
        return P("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", dt(e)), Me(e);
    }
    var me = k.ReactCurrentOwner, vt = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ve, We, $e;
    $e = {};
    function mt(e) {
      if (Z.call(e, "ref")) {
        var t = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (t && t.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function ht(e) {
      if (Z.call(e, "key")) {
        var t = Object.getOwnPropertyDescriptor(e, "key").get;
        if (t && t.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function yt(e, t) {
      if (typeof e.ref == "string" && me.current && t && me.current.stateNode !== t) {
        var r = L(me.current.type);
        $e[r] || (P('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', L(me.current.type), e.ref), $e[r] = !0);
      }
    }
    function Et(e, t) {
      {
        var r = function() {
          Ve || (Ve = !0, P("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", t));
        };
        r.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: r,
          configurable: !0
        });
      }
    }
    function bt(e, t) {
      {
        var r = function() {
          We || (We = !0, P("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", t));
        };
        r.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: r,
          configurable: !0
        });
      }
    }
    var gt = function(e, t, r, l, h, E, p) {
      var f = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: i,
        // Built-in properties that belong on the element
        type: e,
        key: t,
        ref: r,
        props: p,
        // Record the component responsible for creating this element.
        _owner: E
      };
      return f._store = {}, Object.defineProperty(f._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(f, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: l
      }), Object.defineProperty(f, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: h
      }), Object.freeze && (Object.freeze(f.props), Object.freeze(f)), f;
    };
    function Rt(e, t, r, l, h) {
      {
        var E, p = {}, f = null, F = null;
        r !== void 0 && (Ye(r), f = "" + r), ht(t) && (Ye(t.key), f = "" + t.key), mt(t) && (F = t.ref, yt(t, h));
        for (E in t)
          Z.call(t, E) && !vt.hasOwnProperty(E) && (p[E] = t[E]);
        if (e && e.defaultProps) {
          var C = e.defaultProps;
          for (E in C)
            p[E] === void 0 && (p[E] = C[E]);
        }
        if (f || F) {
          var w = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          f && Et(p, w), F && bt(p, w);
        }
        return gt(e, f, F, h, l, me.current, p);
      }
    }
    var Ae = k.ReactCurrentOwner, Ue = k.ReactDebugCurrentFrame;
    function le(e) {
      if (e) {
        var t = e._owner, r = te(e.type, e._source, t ? t.type : null);
        Ue.setExtraStackFrame(r);
      } else
        Ue.setExtraStackFrame(null);
    }
    var ke;
    ke = !1;
    function De(e) {
      return typeof e == "object" && e !== null && e.$$typeof === i;
    }
    function Be() {
      {
        if (Ae.current) {
          var e = L(Ae.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function _t(e) {
      {
        if (e !== void 0) {
          var t = e.fileName.replace(/^.*[\\\/]/, ""), r = e.lineNumber;
          return `

Check your code at ` + t + ":" + r + ".";
        }
        return "";
      }
    }
    var qe = {};
    function St(e) {
      {
        var t = Be();
        if (!t) {
          var r = typeof e == "string" ? e : e.displayName || e.name;
          r && (t = `

Check the top-level render call using <` + r + ">.");
        }
        return t;
      }
    }
    function Ge(e, t) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var r = St(t);
        if (qe[r])
          return;
        qe[r] = !0;
        var l = "";
        e && e._owner && e._owner !== Ae.current && (l = " It was passed a child from " + L(e._owner.type) + "."), le(e), P('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', r, l), le(null);
      }
    }
    function He(e, t) {
      {
        if (typeof e != "object")
          return;
        if (je(e))
          for (var r = 0; r < e.length; r++) {
            var l = e[r];
            De(l) && Ge(l, t);
          }
        else if (De(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var h = N(e);
          if (typeof h == "function" && h !== e.entries)
            for (var E = h.call(e), p; !(p = E.next()).done; )
              De(p.value) && Ge(p.value, t);
        }
      }
    }
    function Pt(e) {
      {
        var t = e.type;
        if (t == null || typeof t == "string")
          return;
        var r;
        if (typeof t == "function")
          r = t.propTypes;
        else if (typeof t == "object" && (t.$$typeof === b || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        t.$$typeof === c))
          r = t.propTypes;
        else
          return;
        if (r) {
          var l = L(t);
          ct(r, e.props, "prop", l, e);
        } else if (t.PropTypes !== void 0 && !ke) {
          ke = !0;
          var h = L(t);
          P("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", h || "Unknown");
        }
        typeof t.getDefaultProps == "function" && !t.getDefaultProps.isReactClassApproved && P("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Ct(e) {
      {
        for (var t = Object.keys(e.props), r = 0; r < t.length; r++) {
          var l = t[r];
          if (l !== "children" && l !== "key") {
            le(e), P("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", l), le(null);
            break;
          }
        }
        e.ref !== null && (le(e), P("Invalid attribute `ref` supplied to `React.Fragment`."), le(null));
      }
    }
    function Je(e, t, r, l, h, E) {
      {
        var p = Re(e);
        if (!p) {
          var f = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (f += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var F = _t(h);
          F ? f += F : f += Be();
          var C;
          e === null ? C = "null" : je(e) ? C = "array" : e !== void 0 && e.$$typeof === i ? (C = "<" + (L(e.type) || "Unknown") + " />", f = " Did you accidentally export a JSX literal instead of a component?") : C = typeof e, P("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", C, f);
        }
        var w = Rt(e, t, r, h, E);
        if (w == null)
          return w;
        if (p) {
          var V = t.children;
          if (V !== void 0)
            if (l)
              if (je(V)) {
                for (var ue = 0; ue < V.length; ue++)
                  He(V[ue], e);
                Object.freeze && Object.freeze(V);
              } else
                P("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              He(V, e);
        }
        return e === d ? Ct(w) : Pt(w), w;
      }
    }
    function xt(e, t, r) {
      return Je(e, t, r, !0);
    }
    function wt(e, t, r) {
      return Je(e, t, r, !1);
    }
    var Tt = wt, Ot = xt;
    ye.Fragment = d, ye.jsx = Tt, ye.jsxs = Ot;
  }()), ye;
}
process.env.NODE_ENV === "production" ? Ie.exports = Dt() : Ie.exports = Ft();
var It = Ie.exports;
const be = It.jsx;
function Te() {
  return Te = Object.assign ? Object.assign.bind() : function(a) {
    for (var i = 1; i < arguments.length; i++) {
      var u = arguments[i];
      for (var d in u)
        Object.prototype.hasOwnProperty.call(u, d) && (a[d] = u[d]);
    }
    return a;
  }, Te.apply(this, arguments);
}
function zt(a, i) {
  typeof a == "function" ? a(i) : a != null && (a.current = i);
}
function Nt(...a) {
  return (i) => a.forEach(
    (u) => zt(u, i)
  );
}
const Oe = /* @__PURE__ */ ge((a, i) => {
  const { children: u, ...d } = a, R = Ee.toArray(u), S = R.find(Mt);
  if (S) {
    const x = S.props.children, T = R.map((b) => b === S ? Ee.count(x) > 1 ? Ee.only(null) : /* @__PURE__ */ xe(x) ? x.props.children : null : b);
    return /* @__PURE__ */ Fe(ze, Te({}, d, {
      ref: i
    }), /* @__PURE__ */ xe(x) ? /* @__PURE__ */ tt(x, void 0, T) : null);
  }
  return /* @__PURE__ */ Fe(ze, Te({}, d, {
    ref: i
  }), u);
});
Oe.displayName = "Slot";
const ze = /* @__PURE__ */ ge((a, i) => {
  const { children: u, ...d } = a;
  return /* @__PURE__ */ xe(u) ? /* @__PURE__ */ tt(u, {
    ...Yt(d, u.props),
    ref: i ? Nt(i, u.ref) : u.ref
  }) : Ee.count(u) > 1 ? Ee.only(null) : null;
});
ze.displayName = "SlotClone";
const Lt = ({ children: a }) => /* @__PURE__ */ Fe(jt, null, a);
function Mt(a) {
  return /* @__PURE__ */ xe(a) && a.type === Lt;
}
function Yt(a, i) {
  const u = {
    ...i
  };
  for (const d in i) {
    const R = a[d], S = i[d];
    /^on[A-Z]/.test(d) ? R && S ? u[d] = (...T) => {
      S(...T), R(...T);
    } : R && (u[d] = R) : d === "style" ? u[d] = {
      ...R,
      ...S
    } : d === "className" && (u[d] = [
      R,
      S
    ].filter(Boolean).join(" "));
  }
  return {
    ...a,
    ...u
  };
}
const ot = rt(void 0), Xt = () => {
  const a = nt(ot);
  if (!a)
    throw new Error("useResplitContext must be used within a ResplitRoot");
  return a;
}, at = rt(void 0), it = () => {
  const a = nt(at);
  if (!a)
    throw new Error("useRootContext must be used within an RootContext.Provider");
  return a;
}, Vt = {
  horizontal: "gridTemplateColumns",
  vertical: "gridTemplateRows"
}, st = {
  horizontal: "col-resize",
  vertical: "row-resize"
}, Wt = "10px", Ut = "0fr", Bt = "0fr", re = (a) => Number(a.replace("fr", "")), lt = (a) => Number(a.replace("px", "")), qt = (a, i) => `${a / i}fr`, ut = (a) => a.includes("px"), Ce = (a, i) => ut(a) ? qt(lt(a), i) : a, Le = typeof window > "u" ? Ne : $t;
function Gt(a) {
  return (i) => {
    a.forEach((u) => {
      typeof u == "function" ? u(i) : u != null && (u.current = i);
    });
  };
}
const Ht = ge(function({ direction: i = "horizontal", children: u, style: d, asChild: R = !1, ...S }, x) {
  const T = At(), b = R ? Oe : "div", _ = we(null), y = we(null), [c, I] = kt({}), g = j(
    (o) => {
      var n;
      return (n = y.current) == null ? void 0 : n.querySelector(`:scope > [data-resplit-order="${o}"]`);
    },
    []
  ), U = j(
    (o) => {
      var n;
      return (n = y.current) == null ? void 0 : n.style.getPropertyValue(`--resplit-${o}`);
    },
    []
  ), G = j(
    (o) => {
      const n = U(o);
      return n ? ut(n) ? lt(n) : re(n) : 0;
    },
    [U]
  ), N = j(
    (o, n) => {
      var v;
      if ((v = y.current) == null || v.style.setProperty(`--resplit-${o}`, n), c[o].type === "pane") {
        const m = g(o + 1);
        m == null || m.setAttribute(
          "aria-valuenow",
          String(re(n).toFixed(2))
        );
      }
    },
    [c, g]
  ), k = j(
    (o) => {
      var n;
      return ((n = g(o)) == null ? void 0 : n.getAttribute("data-resplit-is-min")) === "true";
    },
    [g]
  ), P = j(
    (o, n) => {
      var s;
      return (s = g(o)) == null ? void 0 : s.setAttribute("data-resplit-is-min", String(n));
    },
    [g]
  ), ne = j(
    (o) => {
      var n;
      return ((n = g(o)) == null ? void 0 : n.getAttribute("data-resplit-is-collapsed")) === "true";
    },
    [g]
  ), X = j(
    (o, n) => {
      var s;
      return (s = g(o)) == null ? void 0 : s.setAttribute("data-resplit-is-collapsed", String(n));
    },
    [g]
  ), oe = j(
    () => {
      var o, n;
      return (i === "horizontal" ? (o = y.current) == null ? void 0 : o.offsetWidth : (n = y.current) == null ? void 0 : n.offsetHeight) || 0;
    },
    [i]
  ), ce = j(
    (o, n) => {
      let s = o, v = c[s];
      for (; s >= 0 && s < Object.values(c).length; ) {
        const m = c[s];
        if (m.type === "splitter" || k(s) && !m.options.current.collapsible || k(s) && m.options.current.collapsible && ne(s))
          s += n, v = null;
        else {
          v = m;
          break;
        }
      }
      return { index: s, pane: v };
    },
    [c, ne, k]
  ), W = j(
    (o, n) => {
      var pe, ve;
      const s = n > 0, v = n < 0, { index: m, pane: D } = v ? ce(o - 1, -1) : { index: o - 1, pane: c[o - 1] }, { index: O, pane: M } = s ? ce(o + 1, 1) : { index: o + 1, pane: c[o + 1] };
      if (!D || !M)
        return;
      const A = oe(), Y = D.options.current;
      let $ = G(m) + n;
      const K = re(Ce(Y.minSize, A)), ee = $ <= K, de = !!Y.collapsible && $ <= K / 2, H = M.options.current;
      let z = G(O) - n;
      const se = re(Ce(H.minSize, A)), te = z <= se, Z = !!H.collapsible && z <= se / 2;
      if (de || Z) {
        if (de) {
          const q = re(
            Ce(Y.collapsedSize, A)
          );
          z = z + $ - q, $ = q;
        }
        if (Z) {
          const q = re(
            Ce(H.collapsedSize, A)
          );
          $ = $ + z - q, z = q;
        }
      } else
        ee && (z = z + ($ - K), $ = K), te && ($ = $ + (z - se), z = se);
      N(m, `${$}fr`), P(m, ee), X(m, de), (pe = Y.onResize) == null || pe.call(Y, `${$}fr`), N(O, `${z}fr`), P(O, te), X(O, Z), (ve = H.onResize) == null || ve.call(H, `${z}fr`);
    },
    [
      c,
      ce,
      G,
      oe,
      N,
      P,
      X
    ]
  ), ae = j(
    (o) => {
      if (_.current === null)
        return;
      const n = g(_.current);
      if (!n)
        return;
      const s = Object.entries(c).reduce(
        (M, [A, Y]) => M + (Y.type === "splitter" ? G(Number(A)) : 0),
        0
      ), v = oe() - s, m = n.getBoundingClientRect(), O = (i === "horizontal" ? o.clientX - m.left : o.clientY - m.top) / v;
      O && W(_.current, O);
    },
    [c, i, g, G, oe, W]
  ), ie = j(() => {
    var v, m, D, O, M, A;
    const o = _.current;
    if (o === null)
      return;
    (v = y.current) == null || v.setAttribute("data-resplit-resizing", "false"), o !== null && ((m = g(o)) == null || m.setAttribute("data-resplit-active", "false"));
    const n = c[o - 1];
    n.type === "pane" && ((O = (D = n.options.current).onResizeEnd) == null || O.call(D, U(o - 1)));
    const s = c[o + 1];
    s.type === "pane" && ((A = (M = s.options.current).onResizeEnd) == null || A.call(M, U(o + 1))), _.current = null, document.documentElement.style.cursor = "", document.documentElement.style.pointerEvents = "", document.documentElement.style.userSelect = "", window.removeEventListener("mouseup", ie), window.removeEventListener("mousemove", ae);
  }, [c, g, U, ae]), Re = j(
    (o) => () => {
      var v, m, D, O, M, A;
      _.current = o, (v = y.current) == null || v.setAttribute("data-resplit-resizing", "true"), _.current !== null && ((m = g(_.current)) == null || m.setAttribute("data-resplit-active", "true"));
      const n = c[o - 1];
      n.type === "pane" && ((O = (D = n.options.current).onResizeStart) == null || O.call(D));
      const s = c[o + 1];
      s.type === "pane" && ((A = (M = s.options.current).onResizeStart) == null || A.call(M)), document.documentElement.style.cursor = st[i], document.documentElement.style.pointerEvents = "none", document.documentElement.style.userSelect = "none", window.addEventListener("mouseup", ie), window.addEventListener("mousemove", ae);
    },
    [i, c, g, ie, ae]
  ), _e = j(
    (o) => (n) => {
      const s = i === "horizontal", v = i === "vertical";
      if (n.key === "ArrowLeft" && s || n.key === "ArrowUp" && v)
        W(o, -0.01);
      else if (n.key === "ArrowRight" && s || n.key === "ArrowDown" && v)
        W(o, 0.01);
      else if (n.key === "Home")
        W(o, -1);
      else if (n.key === "End")
        W(o, 1);
      else if (n.key === "Enter")
        if (k(o - 1)) {
          const m = c[o - 1].options.current.initialSize || "1fr";
          W(o, re(m));
        } else
          W(o, -1);
    },
    [i, c, W, k]
  ), fe = j(
    (o, n) => {
      I((s) => ({
        ...s,
        [o]: {
          type: "pane",
          options: n
        }
      }));
    },
    []
  ), L = j(
    (o, n) => {
      I((s) => ({
        ...s,
        [o]: {
          type: "splitter",
          options: n
        }
      }));
    },
    []
  ), B = j(
    (o) => {
      o.forEach((n, s) => {
        var D, O;
        const v = s * 2;
        N(v, n), P(
          v,
          c[v].options.current.minSize === n
        ), X(
          v,
          c[v].options.current.collapsedSize === n
        );
        const m = c[v];
        m.type === "pane" && ((O = (D = m.options.current).onResize) == null || O.call(D, n));
      });
    },
    [c, N, P, X]
  ), Q = Object.keys(c).length;
  Le(() => {
    const o = Object.values(c).filter((n) => n.type === "pane").length;
    Object.keys(c).forEach((n) => {
      const s = Number(n), v = c[s];
      if (v.type === "pane") {
        const m = k(s) ? "0fr" : v.options.current.initialSize || `${1 / o}fr`;
        N(s, m);
      } else
        N(s, v.options.current.size);
    });
  }, [Q]);
  const Se = Ze(
    () => ({
      id: T,
      direction: i,
      registerPane: fe,
      registerSplitter: L,
      handleSplitterMouseDown: Re,
      handleSplitterKeyDown: _e
    }),
    [T, i, fe, L, Re, _e]
  ), Pe = Ze(
    () => ({
      isPaneMinSize: k,
      isPaneCollapsed: ne,
      setPaneSizes: B
    }),
    [k, ne, B]
  );
  return /* @__PURE__ */ be(at.Provider, { value: Se, children: /* @__PURE__ */ be(ot.Provider, { value: Pe, children: /* @__PURE__ */ be(
    b,
    {
      ref: Gt([y, x]),
      "data-resplit-direction": i,
      "data-resplit-resizing": !1,
      style: {
        display: "grid",
        overflow: "hidden",
        [Vt[i]]: Object.keys(c).reduce(
          (o, n) => {
            const s = `minmax(0, var(--resplit-${n}))`;
            return o ? `${o} ${s}` : `${s}`;
          },
          ""
        ),
        ...d
      },
      ...S,
      children: u
    }
  ) }) });
}), Jt = ge(function({
  children: i,
  order: u,
  minSize: d = Ut,
  collapsible: R = !1,
  collapsedSize: S = Bt,
  initialSize: x,
  asChild: T = !1,
  onResize: b,
  onResizeStart: _,
  onResizeEnd: y,
  ...c
}, I) {
  const g = T ? Oe : "div", { id: U, registerPane: G } = it(), N = we({
    minSize: d,
    initialSize: x,
    collapsedSize: S,
    collapsible: R,
    onResize: b,
    onResizeStart: _,
    onResizeEnd: y
  });
  return Le(() => {
    G(String(u), N);
  }, []), Ne(() => {
    N.current = {
      minSize: d,
      initialSize: x,
      collapsedSize: S,
      collapsible: R,
      onResize: b,
      onResizeStart: _,
      onResizeEnd: y
    };
  }, [d, x, S, R, b, _, y]), /* @__PURE__ */ be(
    g,
    {
      id: `resplit-${U}-${u}`,
      "data-resplit-order": u,
      "data-resplit-is-min": !1,
      "data-resplit-is-collapsed": !1,
      ref: I,
      ...c,
      children: i
    }
  );
}), Kt = ge(function({ children: i, order: u, size: d = Wt, asChild: R = !1, ...S }, x) {
  const T = R ? Oe : "div", { id: b, direction: _, registerSplitter: y, handleSplitterMouseDown: c, handleSplitterKeyDown: I } = it(), g = we({ size: d });
  return Le(() => {
    y(String(u), g);
  }, []), Ne(() => {
    g.current = { size: d };
  }, [d]), // eslint-disable-next-line jsx-a11y/role-supports-aria-props, jsx-a11y/no-noninteractive-element-interactions
  /* @__PURE__ */ be(
    T,
    {
      role: "separator",
      tabIndex: 0,
      "aria-orientation": _,
      "aria-valuemin": 0,
      "aria-valuemax": 1,
      "aria-valuenow": 1,
      "aria-controls": `resplit-${b}-${u - 1}`,
      "data-resplit-order": u,
      "data-resplit-active": !1,
      style: { cursor: st[_] },
      onMouseDown: c(u),
      onKeyDown: I(u),
      ref: x,
      ...S,
      children: i
    }
  );
}), Qt = {
  Root: Ht,
  Pane: Jt,
  Splitter: Kt
};
export {
  Qt as Resplit,
  Jt as ResplitPane,
  Ht as ResplitRoot,
  Kt as ResplitSplitter,
  Xt as useResplitContext
};
